aria/workflow_engine/engine.py

```diff
@@ -68,57 +68,71 @@ def create_workflow(
-            {})(context=workflow_context, depends_on=workflow_requires)
-
-    def _create_operation_task(self, graph, operation_context):
-        self.workflow_tasks[operation_context.name] = type(
+        # import pydevd
+        # pydevd.settrace(stderrToServer=True, stdoutToServer=True, suspend=False)
+
+        workflow_start_task = self.workflow_tasks[_start_task_name(graph.name)] = type(
+            _start_task_name(graph.name),
+            (start_task_wrapper,),
+            {})(context=workflow_context, depends_on=depends_on)
+
+        for task_or_workflow in graph.tasks_in_order(reverse=True):
+            dependencies = self._get_dependencies_from_graph(graph, task_or_workflow)
+
+            # Enables to create a workflow framework over the graph
+            dependencies.append(workflow_start_task)
+
+            if _is_operation_task(task_or_workflow):
+                self._create_operation_task(task_or_workflow, dependencies)
+            else:
+                self.create_workflow(
+                    log_indentation_count=log_indentation_count + 1,
+                    graph=task_or_workflow,
+                    workflow_context=workflow_context,
+                    start_task_wrapper=StartSubWorkflowTask,
+                    end_task_wrapper=EndSubWorkflowTask,
+                    depends_on=dependencies)
+
+        workflow_tasks = self._workflow_tasks(graph.tasks)
+        # Every workflow has to have a dependency on the start of the workflow itself
+        workflow_tasks.append(workflow_start_task)
+
+        self.workflow_tasks[_end_task_name(graph.name)] = type(
+            _end_task_name(graph.name),
+            (end_task_wrapper,),
+            {})(context=workflow_context, depends_on=workflow_tasks)
+
+    def _create_operation_task(self, operation_context, depends_on):
+        self.workflow_tasks[_end_task_name(operation_context.name)] = type(
             operation_context.name,
             (OperationTask,),
             {})(context=operation_context,
-                depends_on=self._operation_requires(graph, operation_context),
+                depends_on=depends_on,
                 **operation_context.engine_options)
 
-    def _task_is_operation(self, task):
-        return isinstance(task, OperationContext)
-
-    def _operation_requires(self, graph, context):
+    def _get_dependencies_from_graph(self, graph, task):
         return [
-            self.workflow_tasks[context_dependency.name]
-            for context_dependency in graph.task_dependencies(context)
+            self.workflow_tasks[_end_task_name(dependent_task.name)]
+            for dependent_task in graph.task_dependencies(task)
         ]
 
-    def _workflow_requires(self, contexts):
-        return [
-            self.workflow_tasks[context.name]
-            for context in contexts
-        ]
+    def _workflow_tasks(self, tasks):
+        return [self.workflow_tasks[_end_task_name(task.name)] for task in tasks]
```

Comment:
if task is workflow then use _end_task_name
if task is operation we don't need to add _end_task_name